src/test/java/core/basesyntax/service/impl/FruitTransactionServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionService;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.FruitTransactionStrategy;
+import core.basesyntax.strategy.FruitTransactionStrategyImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionServiceTest {
+    private static FruitTransactionService fruitTransactionService;
+    private static FruitTransactionStrategy strategy;
+    private static Storage fruitsStorage;
+    private static Map<FruitTransaction.Operation,OperationHandler> strategyMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fruitsStorage = new Storage();
+        strategyMap = new HashMap<>();
+        strategyMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        strategyMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        strategyMap.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        strategyMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
+        strategy = new FruitTransactionStrategyImpl(strategyMap);
+        fruitTransactionService = new FruitTransactionServiceImpl(strategy,fruitsStorage);
+    }
+
+    @Test
+    public void processTransaction_Ok() {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        FruitTransaction fruitTransaction = createFruitTransaction("banana", 20,
+                FruitTransaction.Operation.BALANCE);
+        transactions.add(fruitTransaction);
+        FruitTransaction fruitTransaction1 = createFruitTransaction("apple", 100,
```

Comment:
```suggestion
        FruitTransaction firstFruitTransaction = createFruitTransaction("apple", 100,
```
the same for other similar variables